catppuccin-sddm: init at unstable-2023-08-08

GiggleSquid opened the pull request:
## Description of changes
Added the catppuccin SDDM theme
https://github.com/catppuccin/sddm

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

keenanweaver commented:
Result of `nixpkgs-review pr 255808` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>catppuccin-sddm</li>
  </ul>
</details>

keenanweaver reviewed (commented):
Building and working for me.

Instead of downloading all of them, I think it would be nice if you could specify the theme like you can in these other catppuccin SDDM PRs: https://github.com/NixOS/nixpkgs/pull/234800 https://github.com/NixOS/nixpkgs/pull/240990

Example: https://github.com/NixOS/nixpkgs/blob/6bd255dee330bf4b198a7b98d8bf73cd374ee920/pkgs/data/themes/catppuccin-sddm/default.nix#L4

The other options you have are great. I just think it would be best if there was a melding of all three of these PRs!

GiggleSquid committed:
catppuccin-sddm: init at unstable-2023-08-08

GiggleSquid commented:
Bloody hell, after a debacle with rebasing to squish and accidentally picking commits.

This is a great idea, I don't know why I didn't think to do that

> Instead of downloading all of them, I think it would be nice if you could specify the theme like you can in these other catppuccin SDDM PRs: https://github.com/NixOS/nixpkgs/pull/234800 https://github.com/NixOS/nixpkgs/pull/240990


The flavor selection option has been implemented, I've also updated the long description with the changes.

Also fixed the typo you pointed out and removed unstable from the pname

I'm ngl, I never actually checked if there were existing PRs for this. That's my bad, but I think this one has taken good things from the other two, and also the theme config from [sent's PR for the Chili SDDM](https://github.com/NixOS/nixpkgs/pull/224352) theme was the something that i beleive was missing from the other two catppuccin-sddm PRs

keenanweaver commented on pkgs/data/themes/catppuccin-sddm/default.nix:
```diff
@@ -0,0 +1,98 @@
+
+lib.checkListOfEnum "${pname}: flavor" validFlavors [ flavor ]
+
+stdenv.mkDerivation {
```
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

keenanweaver commented on pkgs/data/themes/catppuccin-sddm/default.nix:
````diff
@@ -0,0 +1,98 @@
+       ```
+    '';
+  };
+}
````
```suggestion
})
```

keenanweaver commented:
> I'm ngl, I never actually checked if there were existing PRs for this. That's my bad, but I think this one has taken good things from the other two, and also the theme config from [sent's PR for the Chili SDDM theme](https://github.com/NixOS/nixpkgs/pull/224352) was the something that i beleive was missing from the other two catppuccin-sddm PRs

No problem! Yeah, I think this one is the ultimate PR at this point.

keenanweaver reviewed (commented):
Just something to future-proof. Should build fine.

GiggleSquid commented:
>Just something to future-proof

finalAttrs is another good shout. I've implemented this change

wegank reviewed (dismissed):
Did you forget to add `callPackage` somewhere?

GiggleSquid commented:
> Did you forget to add `callPackage` somewhere?

I must have either lost it in my previous rebase debacle, or I initially forgot to track it!

Added to all-pkgs

GiggleSquid commented:
Closed as #296682 merged
